perllib/FixMyStreet/App/Form/Waste/Garden.pm

```diff
@@ -76,6 +92,13 @@ has_page summary => (
         my $new_bins = $bin_count - $current_bins;
         my $cost_pa = $form->{c}->cobrand->garden_waste_cost_pa($bin_count);
         my $cost_now_admin = $form->{c}->cobrand->garden_waste_new_bin_admin_fee($new_bins);
+        if ($data->{apply_discount} && $form->{c}->stash->{waste_features}->{ggw_discount_pay_proportion}) {
+            my $proportion_to_pay = $form->{c}->stash->{waste_features}->{ggw_discount_pay_proportion};
+            $cost_pa = ($cost_pa / 100) * $proportion_to_pay;
+            $form->{c}->stash->{per_bin_cost} = ($form->{c}->stash->{per_bin_cost} / 100) * $proportion_to_pay;
+            $cost_now_admin = ($cost_now_admin / 100) * $proportion_to_pay;
+        }
```

Comment:
Yes, something like this is more what I'd envisage also for the details_update_fields above - but all moved to a hook that could then be used here as well, presumably.

Reply:
Similarly for modify/renew as well.

Also note, changes here don't actually affect what is then stored as the payment. Which is calculated again in Controller/Waste.pm in process_garden_data/modification/renew. I think if the payment is now going to depend upon more than just the bin count, we need to factor everywhere it's calculated together in one place/hook so we don't miss something.